apps/sports-helsinki/src/domain/search/eventSearch/searchResultList/EventsOrderBySelect.tsx

```diff
@@ -0,0 +1,47 @@
+import { Select, useSearchTranslation } from 'events-helsinki-components';
+import { useRouter } from 'next/router';
+import React from 'react';
+import useHandleEventsOrderChange from '../../../../hooks/useHandleEventsOrderChange';
+import { DEFAULT_EVENT_SORT_OPTION, EVENT_SORT_OPTIONS } from '../constants';
+import styles from './eventsOrderBySelect.module.scss';
+
+const EventsOrderBySelect: React.FC = () => {
+  const { t } = useSearchTranslation();
+  const router = useRouter();
+  const sort = router.query?.sort ?? DEFAULT_EVENT_SORT_OPTION;
+  const handleEventsOrderChange = useHandleEventsOrderChange();
+
+  const orderByOptions = [
+    {
+      text: t('search:orderBy.endTime'),
+      value: EVENT_SORT_OPTIONS.END_TIME,
+    },
+    {
+      text: t('search:orderBy.lastModifiedTime'),
+      value: EVENT_SORT_OPTIONS.LAST_MODIFIED_TIME,
+    },
+    {
+      text: t('search:orderBy.startTime'),
+      value: EVENT_SORT_OPTIONS.START_TIME,
+    },
+  ];
+  const defaultOption = orderByOptions.find((option) => {
+    return option.value === DEFAULT_EVENT_SORT_OPTION;
+  });
+  const selectedOrderByOption = orderByOptions.find((option) => {
+    return option.value === sort;
```

Comment:
That's inside the function given to `find` so I don't think that'd work as intended, I mean if the first value in the orderByOptions would not be equal to router.query?.sort then it would return DEFAULT_EVENT_SORT_OPTION and it'd evaluate as truthy which would make the find function evaluate the first option in the orderByOptions as the selectedOrderByOption, or that's how I understand this code would work. Do you agree or is there something I'm missing here?

Reply:
You're right there is a fine difference... if `router.query?.sort` is not empty but has "wrong value" that doesnt exist on dropdown, then it would use defaultOption. That's the case which is not covered on `option.value === router.query?.sort ?? DEFAULT_EVENT_SORT_OPTION`